src/login/login_auth.cpp

```diff
@@ -273,6 +272,87 @@ int32 login_parse(int32 fd)
                 do_close_login(sd, fd);
             }
             break;
+        case LOGIN_CHANGE_PASSWORD:
+        {
+            const char* fmtQuery = "SELECT accounts.id,accounts.status \
+                                    FROM accounts \
+                                    WHERE accounts.login = '%s' AND accounts.password = PASSWORD('%s')";
+            int32 ret = Sql_Query(SqlHandle, fmtQuery, escaped_name, escaped_pass);
+            if (ret == SQL_ERROR || Sql_NumRows(SqlHandle) == 0)
+            {
+                session[fd]->wdata.resize(1);
+                ref<uint8>(session[fd]->wdata.data(), 0) = LOGIN_ERROR;
+                ShowWarning("login_parse: unexisting user" CL_WHITE"<%s>" CL_RESET" tried to connect\n", escaped_name);
+                do_close_login(sd, fd);
+                return 0;
+            }
+
+            ret = Sql_NextRow(SqlHandle);
+
+            sd->accid = (uint32)Sql_GetUIntData(SqlHandle, 0);
+            uint8 status = (uint8)Sql_GetUIntData(SqlHandle, 1);
+
+            if (status & ACCST_BANNED)
+            {
+                session[fd]->wdata.resize(1);
+                ref<uint8>(session[fd]->wdata.data(), 0) = LOGIN_ERROR_CHANGE_PASSWORD;
+                ShowInfo("login_parse: banned user" CL_WHITE"<%s>" CL_RESET" detected. Aborting.\n", escaped_name);
+                do_close_login(sd, fd);
```

Comment:
Is there no need for a memset like with [during the login flow](https://github.com/project-topaz/topaz/blob/821bcfe6b8db045a03694d032bde530e8152585f/src/login/login_auth.cpp#L165-L170)?

Reply:
It shouldn't be needed since I'm resizing the `wdata` (which is a `std::string`). Per the documentation:

http://www.cplusplus.com/reference/string/string/resize/

> If n is smaller than the current string length, the current value is shortened to its first n character, removing the characters beyond the nth.

So the code resizes the string to a single byte, then we set that byte value to `LOGIN_ERROR_CHANGE_PASSWORD`.

So there's no bytes that contain garbage or need to be copied over with memset.

Reply:
This pattern is used elsewhere in the code as well: https://github.com/project-topaz/topaz/blob/821bcfe6b8db045a03694d032bde530e8152585f/src/login/login_auth.cpp#L97-L98

Reply:
Alright, just curious since I saw a memset was being used for bans later on, and I'm not gonna pretend I'm very familiar with the login server~! 😅

Reply:
As I mentioned in the other thread. A lot of this code can be pulled into helper functions and consolidated for better readability and maintainability. However there's a lot of copy/paste style code blocks throughout here that I didn't wanna mess with for the sake of this change.

A more solid rewrite of this file should be done with care and tested thoroughly.